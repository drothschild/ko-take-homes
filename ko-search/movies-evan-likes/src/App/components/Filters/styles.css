.filters {
  display: block;
  margin: 15px auto auto -0px;
  padding: 0;
  position: relative;
}

.search {
  border-radius: 3px 0 0 3px;
  padding: 5px 8px;
  font: 14px;
  vertical-align: middle;
  width: 200px;
}

.decades {
  background-position: -61px -216px;
  border: medium none;
  color: #999;
  font: 11px Verdana, Arial, sans-serif;
  height: 29px;
  line-height: 25px;
  margin-top: 0px;
  padding: 3px 0 0 6px;
  vertical-align: middle;
  width: 138px;
}
