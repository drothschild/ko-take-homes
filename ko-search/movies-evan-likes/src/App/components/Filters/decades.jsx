import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { setDecadeFilter } from '../Redux/actionCreators'

import styles from './styles.css'

const decadesFromMovies = movieData => {
  const decades = []
  let decadeValues = []
  const defaultValue = (
    <option value="all" key="all">
      All Decades
    </option>
  )
  const years = movieData.map(movie => movie.year)
  years.forEach(year => {
    const decade = Math.floor(year / 10) * 10
    if (decades.indexOf(decade) === -1) {
      decades.push(decade)
    }
  })
  decadeValues = decades.sort().map(decade =>
    /* eslint-disable react/jsx-wrap-multilines */
    <option value={decade} key={decade}>
      {decade}s
    </option>
  )
  decadeValues.unshift(defaultValue)
  return decadeValues
}

const Decades = ({ decadeFilter, handleDecadeChange, movieData }) =>
  /* eslint-disable react/jsx-wrap-multilines */
  <select className={styles.decades} value={decadeFilter} onChange={handleDecadeChange}>
    {decadesFromMovies(movieData)}
  </select>

Decades.propTypes = {
  decadeFilter: PropTypes.string,
  handleDecadeChange: PropTypes.func,
  movieData: PropTypes.arrayOf(PropTypes.object).isRequired
}
Decades.defaultProps = {
  decadeFilter: 'all',
  handleDecadeChange: () => ''
}

const mapStateToProps = state => ({ decadeFilter: state.decadeFilter })
const mapDispatchToProps = dispatch => ({
  handleDecadeChange(event) {
    dispatch(setDecadeFilter(event.target.value))
  }
})

export const Unwrapped = Decades
export default connect(mapStateToProps, mapDispatchToProps)(Decades)
