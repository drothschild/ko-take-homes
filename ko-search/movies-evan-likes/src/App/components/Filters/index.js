import React from 'react'
import styles from './styles.css'

const Filters = () => <div className={styles.filters} />

export default Filters
