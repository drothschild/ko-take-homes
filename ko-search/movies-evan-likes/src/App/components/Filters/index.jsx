import React from 'react'
import PropTypes from 'prop-types'
import Decades from './Decades'
import Search from './Search'

import styles from './styles.css'

const Filters = ({ movieData }) =>
  /* eslint-disable react/jsx-wrap-multilines */
  <div className={styles.filters}>
    <Search />
    <Decades movieData={movieData} />
  </div>

Filters.propTypes = {
  movieData: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default Filters
