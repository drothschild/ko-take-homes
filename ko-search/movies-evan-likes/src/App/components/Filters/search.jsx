import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { setSearchFilter } from '../Redux/actionCreators'

import styles from './styles.css'

const Search = ({ searchFilter, handleSearchTermChange }) =>
  /* eslint-disable react/jsx-wrap-multilines */
  <input
    type="text"
    placeholder="Search"
    value={searchFilter}
    onChange={handleSearchTermChange}
    className={styles.search}
  />

Search.propTypes = {
  searchFilter: PropTypes.string,
  handleSearchTermChange: PropTypes.func
}

Search.defaultProps = {
  searchFilter: '',
  handleSearchTermChange: () => ''
}

const mapStateToProps = state => ({ searchFilter: state.searchFilter })
const mapDispatchToProps = dispatch => ({
  handleSearchTermChange(event) {
    dispatch(setSearchFilter(event.target.value))
  }
})

export const Unwrapped = Search
export default connect(mapStateToProps, mapDispatchToProps)(Search)
