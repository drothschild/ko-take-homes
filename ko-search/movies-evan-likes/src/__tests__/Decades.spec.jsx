import React from 'react'
import { shallow } from 'enzyme'
import { Unwrapped as UnwrappedDecades } from '../App/components/Filters/Decades'
import movieData from '../movieData.json'

test('Decades renders correctly', () => {
  const component = shallow(<UnwrappedDecades movieData={movieData} />)
  expect(component).toMatchSnapshot()
})
