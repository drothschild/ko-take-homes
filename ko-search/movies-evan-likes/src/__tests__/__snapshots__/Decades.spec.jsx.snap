// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decades renders correctly 1`] = `
<select
  onChange={[Function]}
  value="all"
>
  <option
    value="all"
  >
    All Decades
  </option>
  <option
    value={1960}
  >
    1960
    s
  </option>
  <option
    value={1970}
  >
    1970
    s
  </option>
  <option
    value={1980}
  >
    1980
    s
  </option>
  <option
    value={1990}
  >
    1990
    s
  </option>
  <option
    value={2000}
  >
    2000
    s
  </option>
  <option
    value={2010}
  >
    2010
    s
  </option>
</select>
`;
